packages/@aws-cdk/aws-iam/lib/user.ts

```diff
@@ -145,19 +145,13 @@ export class User extends Resource implements IIdentity, IUser {
       public readonly userArn: string = arn;
       public readonly assumeRoleAction: string = 'sts:AssumeRole';
       public readonly policyFragment: PrincipalPolicyFragment = new ArnPrincipal(arn).policyFragment;
-      private defaultPolicy?: Policy;
 
-      public addToPolicy(statement: PolicyStatement): boolean {
-        return this.addToPrincipalPolicy(statement).statementAdded;
+      public addToPolicy(_statement: PolicyStatement): boolean {
+        throw new Error('Cannot add imported User to policy');
```

Comment:
I see no reason to change this method. It will automatically pick up the error from `this.addToPrincipalPolicy`.

Reply:
I agree on this.